### PR TITLE
Added serialize and deserialize to starky proofs

### PR DESCRIPTION
**Pull request by armanthepythonguy**
Added serialize and deserialize to starky proofs. We needed this for our ZKML project where we needed to export the proofs into files and read them back to verify. It might be useful for other projects as well !!!

### CODE REVIEW

File: starky/Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ anyhow = { workspace = true }
 hashbrown = { workspace = true }
 itertools = { workspace = true }
 log = { workspace = true }
+serde = { workspace = true, features = ["rc"] }
```
Comment (Nashtare) [inline starky/Cargo.toml:26]: This is the same import in `plonky2`, it'd be best to import it at the workspace level and then

`serde = { workspace = true }`

**Review by Nashtare — COMMENTED** [review]
Thanks, just a minor nit!

**Review by Nashtare — APPROVED** [review]

**Review by muursh — APPROVED** [review]

**Comment by muursh** [comment]
@armanthepythonguy can you fix the clippy issue then we'll merge

**Comment by Nashtare** [comment]
@armanthepythonguy you need latest nightly toolchain for this to be triggered

**Pull request merged**